File: app/src/main/java/com/kozhevin/rootchecks/annotation/CheckingState.java
package com.kozhevin.rootchecks.annotation;


import androidx.annotation.IntDef;

import java.lang.annotation.Retention;

import static com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_CHECKED_ERROR;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_CHECKED_ROOT_DETECTED;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_CHECKED_ROOT_NOT_DETECTED;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_STILL_GOING;
import static com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_UNCHECKED;
import static java.lang.annotation.RetentionPolicy.SOURCE;



//Why? Just look here https://noobcoderblog.wordpress.com/2015/04/12/java-enum-and-android-intdefstringdef-annotation/
@Retention(SOURCE)
@IntDef({CH_STATE_UNCHECKED,
        CH_STATE_STILL_GOING,
        CH_STATE_CHECKED_ROOT_DETECTED,
        CH_STATE_CHECKED_ROOT_NOT_DETECTED,
        CH_STATE_CHECKED_ERROR,
})
public @interface CheckingState {
}

File: appKotlin/src/main/java/com/kozhevin/rootchecks/annotation/CheckingState.kt
package com.kozhevin.rootchecks.annotation


import androidx.annotation.IntDef
import com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_CHECKED_ERROR
import com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_CHECKED_ROOT_DETECTED
import com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_CHECKED_ROOT_NOT_DETECTED
import com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_STILL_GOING
import com.kozhevin.rootchecks.constant.GeneralConst.CH_STATE_UNCHECKED


//Why? Just look here https://noobcoderblog.wordpress.com/2015/04/12/java-enum-and-android-intdefstringdef-annotation/
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
@IntDef(CH_STATE_UNCHECKED, CH_STATE_STILL_GOING, CH_STATE_CHECKED_ROOT_DETECTED, CH_STATE_CHECKED_ROOT_NOT_DETECTED, CH_STATE_CHECKED_ERROR)
annotation class CheckingState

